src/HOI4World/PlaneDesigns/PlaneDesign.cpp

```diff
@@ -4,71 +4,82 @@
-		type = commonItems::singleString(lambdaStream).getString();
+	registerKeyword("type", [this](std::istream& lambda_stream) {
+		type_ = commonItems::singleString(lambda_stream).getString();
 	});
-	registerKeyword("modules", [this](std::istream& lambdaStream) {
-		modules = std::make_unique<PlaneModules>(lambdaStream);
+	registerKeyword("modules", [this](std::istream& lambda_stream) {
+		modules_ = std::make_unique<PlaneModules>(lambda_stream);
 	});
-	registerKeyword("icon", [this](std::istream& lambdaStream) {
-		icon = commonItems::singleString(lambdaStream).getString();
+	registerKeyword("icon", [this](std::istream& lambda_stream) {
+		icon_ = commonItems::singleString(lambda_stream).getString();
 	});
-	registerKeyword("obsolete", [this](std::istream& lambdaStream) {
-		const commonItems::singleString obsoleteString(lambdaStream);
-		obsolete = obsoleteString.getString() == "yes";
+	registerKeyword("obsolete", [this](std::istream& lambda_stream) {
+		obsolete_ = commonItems::getString(lambda_stream) == "yes";
 	});
-	registerKeyword("required_techs", [this](std::istream& lambdaStream) {
-		for (const auto& requiredTech: commonItems::stringList(lambdaStream).getStrings())
+	registerKeyword("required_techs", [this](std::istream& lambda_stream) {
+		for (const auto& required_tech: commonItems::getStrings(lambda_stream))
 		{
-			requiredTechnologies.insert(requiredTech);
+			required_technologies_.insert(required_tech);
 		}
 	});
-	registerKeyword("blocking_techs", [this](std::istream& lambdaStream) {
-		for (const auto& blockingTech: commonItems::stringList(lambdaStream).getStrings())
+	registerKeyword("blocking_techs", [this](std::istream& lambda_stream) {
+		for (const auto& blocking_tech: commonItems::getStrings(lambda_stream))
 		{
-			blockingTechnologies.insert(blockingTech);
+			blocking_technologies_.insert(blocking_tech);
 		}
 	});
 	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
 
-	parseStream(theStream);
+	parseStream(the_stream);
 	clearRegisteredKeywords();
+
+	if (name_.empty())
+	{
+		throw std::runtime_error("Plane design missing name");
+	}
+	if (type_.empty())
+	{
+		throw std::runtime_error("Plane design missing type");
+	}
 }
 
 
 HoI4::PlaneDesign& HoI4::PlaneDesign::operator=(const PlaneDesign& rhs)
 {
-	name = rhs.name;
-	type = rhs.type;
-	modules = std::make_unique<PlaneModules>(*rhs.modules);
-	icon = rhs.icon;
-	obsolete = rhs.obsolete;
-	requiredTechnologies = rhs.requiredTechnologies;
-	blockingTechnologies = rhs.blockingTechnologies;
+	name_ = rhs.name_;
+	type_ = rhs.type_;
+	modules_ = std::make_unique<PlaneModules>(*rhs.modules_);
+	icon_ = rhs.icon_;
+	obsolete_ = rhs.obsolete_;
+	required_technologies_ = rhs.required_technologies_;
+	blocking_technologies_ = rhs.blocking_technologies_;
 
 	return *this;
 }
 
 
-bool HoI4::PlaneDesign::isValidDesign(const technologies& ownedTechs) const
+bool HoI4::PlaneDesign::IsValidDesign(const technologies& owned_techs) const
```

Comment:
Are techs the only limiting factor in verifying designs? I haven't looked at hoi4 update yet, but presumably an engine module would also be required? Or is this something hoi4 handles on its own akin to marking ships incomplete without some armaments or engines?

Reply:
HoI4 does its own checks to verify designs are valid. This is validity in a converter sense, which is used to avoid providing designs that a country doesn't have tech for or that should be replaced with versions marked obsolete.